lib/ui/profile/Profile.dart

```diff
@@ -348,4 +438,196 @@ class Profile extends StatelessWidget {
       ),
     );
   }
+
+  List<ProfileListItemViewModel> _profileItems(
+    ProfileViewModel viewModel,
+    BuildContext context,
+  ) {
+    List<ProfileListItemViewModel> list = [];
+
+    list.add(ProfileListItemViewModel(
+      title: _LocalisedStrings.switchLanguage(),
+      icon: _Icons.language,
+      onTap: () => _switchLanguage(context, viewModel),
+      isDestructive: false,
+    ));
+
+    list.add(ProfileListItemViewModel(
+      title: _LocalisedStrings.yourFarmDetails(),
+      icon: _Icons.newProfile,
+      onTap: () => _openFarmDetails(viewModel, context),
+      isDestructive: false,
+    ));
+
+    list.add(ProfileListItemViewModel(
+      title: _LocalisedStrings.inviteFriends(),
+      icon: _Icons.inviteFriends,
+      onTap: () => _inviteFriends(),
+      isDestructive: false,
+    ));
+
+    list.add(ProfileListItemViewModel(
+      title: _LocalisedStrings.createNewProfile(),
+      icon: _Icons.soil,
+      onTap: () => _createNewProfile(viewModel, context),
+      isDestructive: false,
+    ));
+
+    list.add(ProfileListItemViewModel(
+      title: _LocalisedStrings.privacyPolicy(),
+      icon: null,
+      onTap: () => _openPrivacyPolicy(context),
+      isDestructive: false,
+    ));
+
+    list.add(ProfileListItemViewModel(
+      title: _LocalisedStrings.termsOfUse(),
+      icon: null,
+      onTap: () => _openTermsOfUse(context),
+      isDestructive: false,
+    ));
+
+    list.add(ProfileListItemViewModel(
+      title: _LocalisedStrings.removeProfile(),
+      icon: null,
+      onTap: () => viewModel.remove(),
+      isDestructive: true,
+    ));
+
+    list.add(ProfileListItemViewModel(
+      title: _LocalisedStrings.logout(),
+      icon: null,
+      onTap: () => viewModel.logout(),
+      isDestructive: true,
+    ));
+    return list;
+  }
+
+  void _switchLanguage(BuildContext context, ProfileViewModel viewModel) {
+    ActionSheet.present(_languageMenu(viewModel), context);
+  }
+
+  void _openFarmDetails(ProfileViewModel viewModel, BuildContext context) {
+    NavigationScope.presentModal(
+      context,
+      FarmDetails(
+        viewModel: FarmDetailsViewModel(
+          items: _mapToFarmItemViewModel(viewModel.farmDetails),
+          buttonTitle: _LocalisedStrings.editFarmDetails(),
+          confirm: () {
+            //TODO Should Navigate to Chat here?
+            viewModel.newAccountFlow.run(context);
```

Comment:
Why this viewModel has references to FlowCoordinator newAccountFlow?
I don't think it should be part of viewModel, but of business logic I guess.

Reply:
it is defined by business logic, and only passed on by the viewmodel. So not too bad. But you are right, this is up for review as its not ideal. But for now it´s the best thing we can do.